nsx/resource_nsx_edge_dlr.go

```diff
@@ -0,0 +1,186 @@
+package nsx
+
+import (
+	"fmt"
+	"github.com/IBM-tfproviders/govnsx"
+	"github.com/IBM-tfproviders/govnsx/nsxresource"
+	"github.com/IBM-tfproviders/govnsx/nsxtypes"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
+
+const (
+	DLRResourceIdPrefix = "dlr-"
+)
+
+type ifCfg struct {
+	name              string
+	ip                string
+	mask              string
+	logical_switch_id string
+}
+
+type dlrCfg struct {
+	edgeId    string
+	ifCfgList []ifCfg
+}
+
+func resourceNsxEdgeDLR() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxEdgeDLRInterfaceCreate,
+		Read:   resourceNsxEdgeDLRInterfaceRead,
+		Update: resourceNsxEdgeDLRInterfaceUpdate,
+		Delete: resourceNsxEdgeDLRInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"edge_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"interface": &schema.Schema{
```

Comment:
We can have one more attribute "type": computed
Any operation on this resource will check the edge type. if it is empty, the resource will be verified as "dlr" with backend call and back populated in the state file. If the type is not dlr, stall the operation and throw an error stating the edge instance is not dlr.